packages/api-client-core/src/GadgetRecordList.ts

```diff
@@ -64,10 +64,12 @@ export class GadgetRecordList<Shape extends RecordShape> extends Array<GadgetRec
   async nextPage() {
     if (!this.hasNextPage)
       throw new GadgetClientError("Cannot request next page because there isn't one, should check 'hasNextPage' to see if it exists");
+    // Our current implementation of paging determines paging direction based on if "first" is defined. We can pass both "before" and "after" as options but only after is respected if first is sent. One of "before" or "after" is ignored depending on whether "first" is defined.
+    const { first, last, before: _before, ...options } = this.pagination.options ?? {};
```

Comment:
I figured I'd extract `before` or `after` so that we don't send both. Our paging implementation won't support sending both anyways